File: add_case.php
<?php
include 'db_connect.php';

if ($_SERVER['REQUEST_METHOD'] !== 'POST') {
    echo "Invalid request";
    exit;
}

$user_id    = $_POST['user_id']    ?? '';
$title      = $_POST['title']      ?? '';
$state      = $_POST['state']      ?? '';
$address    = $_POST['address']    ?? '';
$phone      = $_POST['phone']      ?? '';
$category   = $_POST['category']   ?? '';
$description= $_POST['description']?? '';
$status     = $_POST['status']     ?? 'Pending';

if ($user_id === '' || $title === '' || $state === '' || $address === '' || $phone === '' || $category === '') {
    echo "Please fill required fields";
    exit;
}

$user_id_safe     = mysqli_real_escape_string($conn, $user_id);
$title_safe       = mysqli_real_escape_string($conn, $title);
$state_safe       = mysqli_real_escape_string($conn, $state);
$address_safe     = mysqli_real_escape_string($conn, $address);
$phone_safe       = mysqli_real_escape_string($conn, $phone);
$category_safe    = mysqli_real_escape_string($conn, $category);
$description_safe = mysqli_real_escape_string($conn, $description);
$status_safe      = mysqli_real_escape_string($conn, $status);

$image_path = "";
if (isset($_FILES['image']) && $_FILES['image']['error'] === UPLOAD_ERR_OK) {
    $upload_dir = "uploads/";
    $original_name = basename($_FILES['image']['name']);
    $new_name = time() . "_" . $original_name;
    $target = $upload_dir . $new_name;

    if (move_uploaded_file($_FILES['image']['tmp_name'], $target)) {
        $image_path = $target;
    } else {
        $image_path = "";
    }
}

$sql = "INSERT INTO cases (user_id, title, state, address, phone, category, description, image, status)
        VALUES ('$user_id_safe', '$title_safe', '$state_safe', '$address_safe', '$phone_safe', '$category_safe', '$description_safe', '$image_path', '$status_safe')";

if (mysqli_query($conn, $sql)) {
    echo "Case added successfully";
} else {
    echo "Database error: " . mysqli_error($conn);
}

mysqli_close($conn);
?>

File: add_case.dart
import 'dart:io';
import 'package:flutter/material.dart';
import 'package:image_picker/image_picker.dart';
import 'package:http/http.dart' as http;
import 'package:path/path.dart';

class AddCasePage extends StatefulWidget {
  final int userId;

  const AddCasePage({super.key, required this.userId});

  @override
  State<AddCasePage> createState() => _AddCasePageState();
}

class _AddCasePageState extends State<AddCasePage> {
  final _formKey = GlobalKey<FormState>();

  TextEditingController titleController = TextEditingController();
  TextEditingController addressController = TextEditingController();
  TextEditingController phoneController = TextEditingController();
  TextEditingController descriptionController = TextEditingController();

  String? selectedState;
  String? selectedCategory;
  File? imageFile;

  final List<String> states = [
    'Khartoum',
    'North Darfur',
    'South Darfur',
    'Blue Nile',
    'River Nile',
    'Gezira',
  ];

  final List<String> categories = [
    'Reconstruction',
    'Medical Aid',
    'Education',
    'Food Support',
    'Shelter',
  ];

  Future<void> pickImage() async {
    final picked = await ImagePicker().pickImage(source: ImageSource.gallery);
    if (picked != null) {
      setState(() {
        imageFile = File(picked.path);
      });
    }
  }

  Future<void> submitCase(BuildContext context) async {
    if (!_formKey.currentState!.validate()) return;

    var uri = Uri.parse("http://10.0.2.2/pwrs_backend/add_case.php");

    var request = http.MultipartRequest('POST', uri);
    request.fields['user_id'] = widget.userId.toString();
    request.fields['title'] = titleController.text;
    request.fields['state'] = selectedState ?? '';
    request.fields['address'] = addressController.text;
    request.fields['phone'] = phoneController.text;
    request.fields['category'] = selectedCategory ?? '';
    request.fields['description'] = descriptionController.text;
    request.fields['status'] = 'Pending';

    if (imageFile != null) {
      request.files.add(
        await http.MultipartFile.fromPath(
          'image',
          imageFile!.path,
          filename: basename(imageFile!.path),
        ),
      );
    }

    var response = await request.send();

    if (!mounted) return;

    if (response.statusCode == 200) {
      ScaffoldMessenger.of(
        context,
      ).showSnackBar(const SnackBar(content: Text('Case added successfully')));
      Navigator.pop(context);
    } else {
      ScaffoldMessenger.of(
        context,
      ).showSnackBar(const SnackBar(content: Text('Failed to add case')));
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Report Case'),
        centerTitle: true,
        backgroundColor: Colors.blueAccent,
      ),
      body: SingleChildScrollView(
        padding: const EdgeInsets.all(16.0),
        child: Form(
          key: _formKey,
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              TextFormField(
                controller: titleController,
                decoration: const InputDecoration(labelText: 'Subject'),
                validator: (value) =>
                    value!.isEmpty ? 'Please enter a subject' : null,
              ),
              const SizedBox(height: 12),
              DropdownButtonFormField<String>(
                decoration: const InputDecoration(labelText: 'State'),
                items: states
                    .map((s) => DropdownMenuItem(value: s, child: Text(s)))
                    .toList(),
                onChanged: (value) => setState(() => selectedState = value),
                validator: (value) =>
                    value == null ? 'Please select a state' : null,
              ),
              const SizedBox(height: 12),
              TextFormField(
                controller: addressController,
                decoration: const InputDecoration(labelText: 'Address'),
                validator: (value) =>
                    value!.isEmpty ? 'Please enter address' : null,
              ),
              const SizedBox(height: 12),
              TextFormField(
                controller: phoneController,
                decoration: const InputDecoration(labelText: 'Phone Number'),
                keyboardType: TextInputType.phone,
                validator: (value) =>
                    value!.isEmpty ? 'Please enter phone number' : null,
              ),
              const SizedBox(height: 12),
              DropdownButtonFormField<String>(
                decoration: const InputDecoration(labelText: 'Type of Request'),
                items: categories
                    .map((c) => DropdownMenuItem(value: c, child: Text(c)))
                    .toList(),
                onChanged: (value) => setState(() => selectedCategory = value),
                validator: (value) =>
                    value == null ? 'Please select a category' : null,
              ),
              const SizedBox(height: 12),
              TextFormField(
                controller: descriptionController,
                maxLines: 4,
                decoration: const InputDecoration(labelText: 'Description'),
                validator: (value) =>
                    value!.isEmpty ? 'Please enter description' : null,
              ),
              const SizedBox(height: 16),
              const Text('Attachment'),
              const SizedBox(height: 8),
              GestureDetector(
                onTap: pickImage,
                child: Container(
                  height: 150,
                  width: double.infinity,
                  decoration: BoxDecoration(
                    border: Border.all(color: Colors.grey),
                    borderRadius: BorderRadius.circular(8),
                  ),
                  child: imageFile == null
                      ? const Center(
                          child: Column(
                            mainAxisAlignment: MainAxisAlignment.center,
                            children: [
                              Icon(
                                Icons.upload_file,
                                size: 40,
                                color: Colors.grey,
                              ),
                              Text('Click here to upload image'),
                            ],
                          ),
                        )
                      : Image.file(imageFile!, fit: BoxFit.cover),
                ),
              ),
              const SizedBox(height: 20),
              ElevatedButton(
                onPressed: () {
                  submitCase(context);
                },
                style: ElevatedButton.styleFrom(
                  backgroundColor: Colors.blueAccent,
                  minimumSize: const Size(double.infinity, 50),
                ),
                child: const Text('Submit'),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
